### ISSUE TITLE
arrow-parens new option "omit"

### ISSUE THREAD

**Issue by AndersDJohnson** [eslint/eslint#8946]
**What rule do you want to change?**

[arrow-parens](http://eslint.org/docs/rules/arrow-parens)

**Does this change cause the rule to produce more or fewer warnings?**

More, compared to existing "as-needed" option where parens are optional.

**How will the change be implemented? (New option, new default behavior, etc.)?**

New option.

**Please provide some example code that this change will affect:**

```js
const fn = (e) => false;
```

**What does the rule currently do for this code?**

Allows the parentheses with "as-needed" option, requires them with "always" option.

**What will the rule do after it's changed?**

Will warn in new "omit" option until code is corrected to omit parentheses:

```js
const fn = e => false;
```

**Comment by platinumazure** [comment]
Seems reasonable to me, although I think we would probably call the option `"never"` to be consistent with other rules. :+1: from me.

**Comment by not-an-aardvark** [comment]
I'm confused -- doesn't the `as-needed` option already cover this?

> **What does the rule currently do for this code?**
>
> Allows the parentheses with "as-needed" option, requires them with "always" option.

I don't think that's correct -- the parentheses are disallowed with the `as-needed` option.

**Comment by AndersDJohnson** [comment]
@not-an-aardvark Ah, sorry, you are correct.

We had `eslint-config-prettier` in our project, which disables this rule, and so obscuring this behavior from me.

The docs are worded a bit confusingly, stating "'as-needed' allows omitting parens," which makes them sound optional rather than suggesting it will warn/error if parens are needlessly included.
